test.py

```diff
@@ -0,0 +1,247 @@
+
+    projGenParams.append (str(rootFolder))
+
+    projGenResult = subprocess.call (projGenParams)
+    if projGenResult != 0:
+        print ('Failed to generate project')
+        return 1
+    
+    # Add params to build AddOn
+    buildParams = []
+    buildParams.append ('cmake')
+    buildParams.extend (['--build', str(buildPath)])
+    buildParams.extend (['--config', configuration])
+
+    buildResult = subprocess.call (buildParams)
+    if buildResult != 0:
+        print ('Failed to build project')
+        return 1
+    
+    return 0
+
+def CopyResultToPackage(packageRootFolder, buildFolder, addOnName, platformName, configuration, languageCode=None, isRelease=False):
+    packageFolder = packageRootFolder
+    sourceFolder = buildFolder / addOnName
+
+    if languageCode is not None:
+        packageFolder = packageFolder / languageCode
+        sourceFolder = sourceFolder / languageCode
+    sourceFolder = sourceFolder / configuration
+
+    if not packageFolder.exists():
+        packageFolder.mkdir(parents=True)
+
+    fileName = addOnName
+    if not isRelease:
+        fileName = fileName + '_' + configuration
+
+    if platformName == 'WIN':
+        shutil.copy (
+            sourceFolder / (addOnName + '.apx'),
+            packageFolder / (fileName + '.apx')
+        )
+        if configuration == 'Debug':
+            shutil.copy (
+                sourceFolder / (addOnName + '.pdb'),
+                packageFolder / (fileName + '.pdb')
+            )
+
+    elif platformName == 'MAC':
+        subprocess.call ([
+            'cp', '-r',
+            sourceFolder / (addOnName + '.bundle'),
+            packageFolder / (fileName + '.bundle')
+        ])
+    
+
+
+def Main():
+    parser = argparse.ArgumentParser ()
+    parser.add_argument ('--acVersion', dest = 'acVersion', type = str, required = True, help = 'Archicad version number list. Ex: "26 27"')
+    parser.add_argument ('--language', dest = 'language', type = str, required = False, help = 'Add-On language code list. Ex: "INT GER". Specify "ALL" for all languages in the configfile.' )
+    parser.add_argument ('--devKitPath', dest = 'devKitPath', type = str, required = False, help = 'Path to local APIDevKit')
+    parser.add_argument ('--release', dest = 'release', required = False, action='store_true', help = 'Build in localized Release mode.')
+    parser.add_argument ('--package', dest = 'package', required = False, action='store_true', help = 'Create zip archive.')
+    args = parser.parse_args ()
+
+    if args.language is None and args.release:
+        print('Must specify AddOn language on release version')
+        return 1
+
+    acVersionList = args.acVersion.split()
+        
+    if args.devKitPath is not None and len(acVersionList) != 1:
+        print('Only one Archicad version supported with local APIDevKit option')
+        return 1
+
+    # Check platform operating system
+    platformName = None

```

Comment:
This quite a robust main function. Moving the logic to separate functions would be a better idea.

Reply:
Moved some logic out of the Main function, but the initial verifications and variable assignments are a bit tricky to separate